### ISSUE TITLE
Input lag when typing in the post editor

### ISSUE THREAD

**Issue by AngeloStavrow** [writeas/writefreely-swiftui-multiplatform#43]
After a few words, there's a noticeable (but bearable) lag between when you type a character and when the UI updates to show it. After a couple of sentences, this lag becomes ridiculous.

**Comment by AngeloStavrow** [comment]
Initially I thought this might have something to do with the `.onChange(of:)` modifiers attached to the TextEditor elements in PostEditorView, but that doesn't seem to be the case here.

After spending some time in Instruments I'm seeing that _NSEventThread is getting blocked and then, after about 7 seconds of typing, the main thread starts blocking. The backtrace is mostly Apple's stuff:

```
   0 libsystem_kernel.dylib mach_msg_trap
   1 libsystem_kernel.dylib mach_msg
   2 CoreFoundation __CFRunLoopServiceMachPort
   3 CoreFoundation __CFRunLoopRun
   4 CoreFoundation CFRunLoopRunSpecific
   5 HIToolbox RunCurrentEventLoopInMode
   6 HIToolbox ReceiveNextEventCommon
   7 HIToolbox _BlockUntilNextEventMatchingListInModeWithFilter
   8 AppKit _DPSNextEvent
   9 AppKit -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]
  10 AppKit -[NSApplication run]
  11 AppKit NSApplicationMain
  12 SwiftUI specialized runApp(_:)
  13 SwiftUI runApp<A>(_:)
  14 SwiftUI static App.main()
  15 WriteFreely-MultiPlatform static WriteFreely_MultiPlatformApp.$main() ⚠️
  16 WriteFreely-MultiPlatform main ⚠️
  17 libdyld.dylib start
```

Items 15 and 16 above (annotated with ⚠️ for convenience) refer to  /Shared/Navigation/ContentView.swift:0 — the entry point to the app, so I don't think this is helpful. Seems to me this may be a SwiftUI/macOS bug so I'm going to set this aside until the next release of Xcode/macOS to see if things improve with that.

**Comment by AngeloStavrow** [comment]
In my testing right now, simply commenting out the `.toolbar` modifier and its content in **macOS/PostEditor/PostEditorView.swift** (i.e., the post status badge and publish button) reduces the input lag to zero. While I need to do a bit more testing to explain _why_, right now it looks like the work to resolve this performance issue may involve #128 as well.

**Comment by AngeloStavrow** [comment]
Closing this as `wontfix` due to the fact that it will be corrected in #128.
